compiler/qsc/src/bin/qsc.rs

```diff
@@ -74,12 +75,21 @@ fn main() -> miette::Result<ExitCode> {
         dependencies.push(store.insert(qsc::compile::std(&store, target)));
     }
 
-    let sources = cli
+    let mut sources = cli
         .sources
         .iter()
         .map(read_source)
         .collect::<miette::Result<Vec<_>>>()?;
 
+    let fs = StdFs;
+    let manifest = Manifest::load()?;
+    if let Some(manifest) = manifest {
+        let project = fs.load_project(manifest)?;
+        let mut project_sources = project.sources;
+
+        sources.append(&mut project_sources);
```

Comment:
In the description for the PR (which seems to be cut off for some reason?) the expected behavior is that a manifest is loaded only when no input sources are provided, but this code path looks like it will execute no matter what. Should it check first to ensure the `sources` vector is empty?